content/2016.12.prototyping-an-app-with-kinto.md

```diff
@@ -0,0 +1,145 @@
+Title: Prototyping an app with Kinto
+Date: 2016-12-15
+Lang: en
+Url: prototyping-an-app-with-kinto
+Slug: prototyping-an-app-with-kinto
+Summary: Prototyping an app with Kinto and the admin.
+
+I own a car. And it costs a lot. But how much exactly is difficult to know, because the costs are spread between maintenance, repairs, fuel, insurance...
+What if I could store each expense right on the spot, tag it, and maybe add a description? That would be tremendously helpful, right?
+
+In order to build something like that, I'd normally have to fire up a django project, maybe use django-rest-framework to have a proper REST API for the future web or mobile application. Then choose a database like postgresql and configure it.
+
+But I'm not even sure it's a good idea! Maybe we can instead give it a try using the [kinto-admin].
+
+## Let's get started
+
+Let's first think about the data model:
+
+* title: a string
+* comment: a multiline string
+* tag: a string taken from a list of choices
+* price: a float
+
+And then how to store entries in Kinto:
+
+* we'll have a bucket (`track-my-expenses`) which contains a collection (`car-expenses`)
+* a [JSON schema] on the collection will validate the data before it's stored in records
+* and a [UI schema] to improve the way the form is displayed in the admin (eg. a textarea for the comment instead of a single line text input).
+
+Our proof of concept will take advantage of the [kinto-admin] UI which gives us a way to create/read/update/delete entries (records).
+
+
+## Create the bucket and the collection
+
+For our proof of concept, we'll use Mozilla's demo server, which is wiped clean every night: https://kinto.dev.mozaws.net/v1/.
+
+Let's sign in to the [admin] with the user/pass of our choice (the user will be "created" on the spot):
```

Comment:
> the user will be "created" on the spot

It is not the truth, kinto doesn't handle any kind of user using the default basicauth backend.
If you talk about users people then want to change password for their user, while there is actually no user.